File: typescript/lld/booking/hotel/Reservation.ts
import { DateRange } from './DateRange';
import { ReservationStatus } from './enums';

export class Reservation {
    status: ReservationStatus = ReservationStatus.HELD;
    constructor(
        public readonly id: string,
        public readonly roomNumber: string,
        public readonly guestId: string,
        public readonly range: DateRange,
        public readonly amount: number,
    ) {}
}

File: java/lld/booking/hotel/Reservation.java
public class Reservation {
    public final String id, roomNumber, guestId;
    public final DateRange range;
    public final double amount;
    public ReservationStatus status = ReservationStatus.HELD;
    public Reservation(String id, String roomNumber, String guestId, DateRange range, double amount) {
        this.id = id; this.roomNumber = roomNumber; this.guestId = guestId;
        this.range = range; this.amount = amount;
    }
}